vumi_http_retry/retries.py

```diff
@@ -20,3 +26,34 @@ def add_request(redis, prefix, req):
     }
 
     return redis.zadd(requests_key(prefix), next_score(req), json.dumps(req))
+
+
+@inlineCallbacks
+def peek_requests(redis, prefix, from_time, to_time):
+    k = requests_key(prefix)
+
+    returnValue([
+        json.loads(r)
+        for r in (yield redis.zrangebyscore(k, from_time, to_time))])
+
+
+@inlineCallbacks
+def pop_requests(redis, prefix, from_time, to_time):
+    k = requests_key(prefix)
+    requests = yield peek_requests(redis, prefix, from_time, to_time)
+    yield redis.zremrangebyscore(k, from_time, to_time)
```

Comment:
This has a possibility of a race condition. Maybe we should be using `zrem` with the list we just got?